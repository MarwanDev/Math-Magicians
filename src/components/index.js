export { default as Calculator } from './Calculator/Calculator';
export { default as Quotes } from './Quotes/Quotes';
export { default as Navbar } from './Navbar/Navbar';
export { default as Layout } from './Layout/Layout';
export { default as Home } from './Home/Home';
