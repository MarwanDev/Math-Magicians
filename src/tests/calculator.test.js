import renderer from 'react-test-renderer';
import { Calculator } from '../components';

it('renders calculator page correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
