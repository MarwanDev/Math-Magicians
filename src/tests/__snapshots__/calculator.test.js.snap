// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders calculator page correctly`] = `
  <div className="calculator">
    <div className="calculator-container">
      <h2 className="calc-output">
        {total}
        {operation}
        {next}
      </h2>
      <div className="calculator-btns-container">
        <div className="btn-container">
          <button type="button">AC</button>
          <button type="button">+/-</button>
          <button type="button">%</button>
          <button type="button" className="operator">รท</button>
          <button type="button">7</button>
          <button type="button">8</button>
          <button type="button">9</button>
          <button type="button" className="operator">x</button>
          <button type="button">4</button>
          <button type="button">5</button>
          <button type="button">6</button>
          <button type="button"className="operator">-</button>
          <button type="button">1</button>
          <button type="button">2</button>
          <button type="button">3</button>
          <button type="button" className="operator">+</button>
          <button type="button" className="zero-btn">0</button>
          <button type="button">.</button>
          <button type="button" className="operator">=</button>
        </div>
      </div>
    </div>
  </div>
`;

exports[`renders calculator page correctly 1`] = `
<div
  className="calculator"
>
  <div
    className="calculator-container"
  >
    <h2
      className="calc-output"
    />
    <div
      className="calculator-btns-container"
    >
      <div
        className="btn-container"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="zero-btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="operator"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
